src/main/java/net/sf/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -233,7 +243,8 @@ private boolean saveDatabase(File file, boolean selectedOnly, Charset encoding)
         try {
             if (success) {
                 session.commit(file.toPath());
-                panel.getBibDatabaseContext().getMetaData().setEncoding(encoding); // Make sure to remember which encoding we used.
+                // Make sure to remember which encoding we used.
+                panel.getBibDatabaseContext().getMetaData().setEncoding(encoding, ChangePropagation.DO_NOT_POST_EVENT);
```

Comment:
> Probably you don't want to trigger a new save of the file

It's even worse, this would cause an event loop which would never end.

> the problem with the current solution is that nobody else can listen to encoding changes in a reliable way

There is still the old method `setEncoding(Charset encoding)` which is now a shortcut for `setEncoding(encoding, ChangePropagation.POST_EVENT)`. Every time you set the encoding the event is also going to be posted (which will also trigger the autosave). It's ok so far. But the event is not going to be posted while saving a file. I'm more skaptical why this is called here at all. The old comment `// Make sure to remember which encoding we used.` is not really justified, as the encoding should be set immediately when it changes and not "at the last second".